src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java

```diff
@@ -116,15 +114,16 @@ public StorageAdaptorRetrieveResponse retrieve(@NotBlank final String key)
           MediaType.valueOf(objectMetadata.getContentType()),
           inputStream,
           objectMetadata.getUserMetadata());
-    } catch (final Throwable t) {
+    } catch (final Exception e) {
```

Comment:
I could be wrong, but if anything more severe than Exception gets thrown, the service is already in a non recoverable state. At that point it doesn't matter if the stream is closed or not.

Reply:
Throwable will catch all RuntimeExceptions as well if that was the intention here.

Reply:
Ah yeah, that's a good point. I think that was the intention. I'll revert it.

Reply:
Maybe just catch Exception and RuntimeException separately? We really shouldn't be catching Throwable.